### PR TITLE
spi housekeeping

### PR DESCRIPTION
**Pull request by MabezDev**
## Thank you for your contribution!

We appreciate the time and effort you've put into this pull request.
To help us review it efficiently, please ensure you've gone through the following checklist:

### Submission Checklist 📝
- [x] I have updated existing examples or added new ones (if applicable).
- [x] My changes were added to the [`CHANGELOG.md`](https://github.com/esp-rs/esp-hal/blob/main/esp-hal/CHANGELOG.md) in the **_proper_** section.
#### Extra:
- [x] I have read the [CONTRIBUTING.md guide](https://github.com/esp-rs/esp-hal/blob/main/CONTRIBUTING.md) and followed its instructions.

### CODE REVIEW

File: esp-hal/src/spi/master.rs [hunk 1]
```diff
@@ -806,7 +806,8 @@ where
     type Error = Error;
 
     fn write(&mut self, words: &[u8]) -> Result<(), Self::Error> {
-        self.write_bytes(words)
+        self.write_bytes(words)?;
+        self.spi.flush()
```
Comment (MabezDev) [inline esp-hal/src/spi/master.rs:810]: Removed a fixme by moving adding this flush on eh02 impls

File: esp-hal/src/spi/master.rs [hunk 2]
```diff
@@ -1454,8 +1455,6 @@ pub mod dma {
 
                     crate::dma::asynch::DmaTxFuture::new(&mut self.channel.tx).await;
 
-                    // FIXME: in the future we should use the peripheral DMA status registers to
```
Comment (MabezDev) [inline esp-hal/src/spi/master.rs:1457]: This comments are redundant, imo using dma then waiting for the last few bytes to get flushed is the optimal way.

**Review by jessebraham — APPROVED** [review]
Thanks!

File: esp-hal/src/spi/master.rs [hunk 3]
```diff
@@ -2391,7 +2385,6 @@ pub trait Instance: crate::private::Sealed {
 
     // taken from https://github.com/apache/incubator-nuttx/blob/8267a7618629838231256edfa666e44b5313348e/arch/risc-v/src/esp32c3/esp32c3_spi.c#L496
     fn setup(&mut self, frequency: HertzU32, clocks: &Clocks) {
-        // FIXME: this might not be always true
```
[the remark this answers is not in the document]
Comment (Dominaezzz) [reply esp-hal/src/spi/master.rs:2394]: When you say "managing", do you mean just picking a source clock based on the users desired frequency (and the possible dividers)?

[the remark this answers is not in the document]
Comment (MabezDev) [reply esp-hal/src/spi/master.rs:2394]: Essentially yes, but for most drivers we have that use case covered. What we don't have is any kind of configuration from the user side (if we can get away with it, maybe we shouldn't expose much clock stuff to users), but it's needed in some cases, see https://github.com/esp-rs/esp-hal/issues/1181

[the remark this answers is not in the document]
Comment (Dominaezzz) [reply esp-hal/src/spi/master.rs:2394]: I wrote something like this for the LCD_CAM driver. ([this thing](https://github.com/esp-rs/esp-hal/blob/main/esp-hal/src/lcd_cam/mod.rs#L62)). Might be useful for this case as well then

**Review by bjoernQ — APPROVED** [review]
LGTM

**Pull request merged**